asapdiscovery-modeling/asapdiscovery/modeling/tests/test_protein_prep.py

```diff
@@ -0,0 +1,284 @@
+    def test_sars_protein_prep(
+        self, sars_xtal, ref, prepped_files, loop_db, ref_chain="A"
+    ):
+        xtal = sars_xtal
+
+        # Load structure
+        prot = load_openeye_pdb(xtal.str_fn)
+
+        assert type(prot) == oechem.OEGraphMol
+
+        prot = remove_extra_ligands(prot, lig_chain=xtal.lig_chain)
+
+        # TODO add a test to confirm only a ligand in chain A is there
+
+        # Align to reference
+        prot = align_receptor(
+            prot,
+            dimer=True,
+            ref_prot=str(ref),
+            split_initial_complex=False,
+            split_ref=True,
+            ref_chain=ref_chain,
+            mobile_chain=xtal.active_site_chain,
+            keep_water=False,
+        )
+
+        # TODO add test to make sure the active site is aligned
+
+        save_openeye_pdb(prot, prepped_files / f"{xtal.output_name}_align.pdb")
+
+        # Mutate Residues
+        seqres_yaml = fetch_test_file("mpro_sars2_seqres.yaml")
+
+        with open(seqres_yaml) as f:
+            seqres_dict = yaml.safe_load(f)
+        seqres = seqres_dict["SEQRES"]
+
+        res_list = seqres_to_res_list(seqres)
+
+        prot = mutate_residues(prot, res_list, place_h=True)
+        seqres = " ".join(res_list)
+
+        save_openeye_pdb(prot, prepped_files / f"{xtal.output_name}_mutate.pdb")
+
+        # Spruce Protein
+        du = spruce_protein(
+            initial_prot=prot,
+            seqres=seqres,
+            loop_db=loop_db,
+            return_du=True,
+            site_residue=xtal.active_site,
+        )
+        assert type(du) == oechem.OEDesignUnit
+
+        du_fn = prepped_files / f"{xtal.output_name}-prepped_receptor_0.oedu"
+        oechem.OEWriteDesignUnit(str(du_fn), du)
+
+        # Serialize output!
+        from asapdiscovery.data.openeye import (
+            save_openeye_sdf,
+            split_openeye_design_unit,
+        )
+        from asapdiscovery.modeling.modeling import add_seqres_to_openeye_protein
+
+        # TODO: Use a different way of splitting the design unit
+        lig, prot, complex_ = split_openeye_design_unit(du, lig_title=xtal.compound_id)
+        prot = add_seqres_to_openeye_protein(prot, seqres)
+        complex_ = add_seqres_to_openeye_protein(complex_, seqres)
+
+        prot_fn = prepped_files / f"{xtal.output_name}-prepped_protein.pdb"
+        save_openeye_pdb(prot, str(prot_fn))
+
+        complex_fn = prepped_files / f"{xtal.output_name}-prepped_complex.pdb"
+        save_openeye_pdb(complex_, str(complex_fn))
+
+        lig_fn = prepped_files / f"{xtal.output_name}-prepped_ligand.sdf"
+        save_openeye_sdf(lig, str(lig_fn))
+
+        for fn in [prot_fn, complex_fn, lig_fn]:
+            assert Path(fn).exists()
+            assert Path(fn).is_file()
+
+        # TODO: Add a test to make sure the ligand is in the active site
+        # TODO: Add a test to make sure the seqres has been added to the protein
+
+    def test_mers_protein_prep(
```

Comment:
The sars and mers tests can be merged with a `parametrize()` decorator as they test the same functionality.

Reply:
and `prep_mers` here